clients/lsp-javascript.el

```diff
@@ -817,28 +822,49 @@ name (e.g. `data' variable passed as `data' parameter)."
       (lsp)
       (lsp--info "Renamed '%s' to '%s'." name (file-name-nondirectory new)))))
 
+(defun lsp-javascript-update-inlay-hints-scroll-function (window start)
+  (lsp-javascript--update-inlay-hints start (window-end window t)))
+
 (defun lsp-javascript-update-inlay-hints ()
+  (lsp-javascript--update-inlay-hints (window-start) (window-end nil t)))
+
+(defun lsp-javascript--update-inlay-hints (start end)
   (if (lsp-javascript-initialized?)
       (lsp-request-async
        "typescript/inlayHints"
        (lsp-make-javascript-inlay-hints-params
-        :text-document (lsp--text-document-identifier))
+        :text-document (lsp--text-document-identifier)
+        :range (lsp-make-javascript-range :start
+                                          (lsp-point-to-position start)
+                                          :end
+                                          (lsp-point-to-position end)))
        (lambda (res)
          (lsp--remove-overlays 'lsp-javascript-inlay-hint)
-         (-each (gethash "inlayHints" res)
-           #'(lambda (hint)
-               (-let* (((&javascript:InlayHint :text :position :kind :whitespace-before? :whitespace-after?) hint)
-                       (pos (lsp--position-to-point position))
-                       (overlay (make-overlay pos pos nil 'front-advance 'end-advance)))
-                 (overlay-put overlay 'lsp-javascript-inlay-hint t)
-                 (overlay-put overlay 'before-string
-                              (format "%s%s%s"
-                                      (if (and whitespace-before? (not (string= kind lsp/javascript-inlay-hint-kind-type-hint))) " " "")
-                                      (propertize (lsp-javascript-format-inlay text kind)
-                                                  'font-lock-face (lsp-javascript-face-for-inlay kind))
-                                      (if whitespace-after? " " ""))))))))
+         (let ((hints (lsp-get res :inlayHints)))
+           (when (> (length hints) 0)
```

Comment:
Open to suggestions for how to phrase this entire `when` block. The idea is to call `overlay-recenter` only if there are hints to add.

Reply:
looks ok to me, you may use `(unelss (seq-empty-p...`.